### PR TITLE
treewide: remove hardcoded /build

### PR DESCRIPTION
**Pull request by pbsds**
## Description of changes

grepping for ` /build` reveals a few demons, some of which i exorcise here

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by dasj19 — APPROVED** [review]
Looks good

File: pkgs/development/python-modules/triton/default.nix [hunk 1]
```diff
@@ -178,7 +178,7 @@ buildPythonPackage rec {
 
   preCheck = ''
     # build/temp* refers to build_ext.build_temp (looked up in the build logs)
-    (cd /build/source/python/build/temp* ; ctest)
+    (cd ./build/temp* ; ctest)
```
Comment (Atemu) [inline pkgs/development/python-modules/triton/default.nix:181]: Have you verified that this actually works on locations other than /build and isn't just relative to `/`?

File: pkgs/applications/video/kodi/addons/jellycon/default.nix [hunk 2]
```diff
@@ -29,7 +29,7 @@ buildKodiAddon rec {
   '';
 
   postInstall = ''
-    mv /build/source/addon.xml $out${addonDir}/${namespace}/
+    cp -v addon.xml $out${addonDir}/$namespace/
```
Comment (Atemu) [inline pkgs/applications/video/kodi/addons/jellycon/default.nix:32]: `install`?

**Review by Atemu — COMMENTED** [review]
Diff generally looks good to me.

File: pkgs/applications/video/kodi/addons/jellycon/default.nix (hunk 2 above)
Comment (pbsds) [reply pkgs/applications/video/kodi/addons/jellycon/default.nix:32]: due to `-t` i find `install` way harder to read than `cp` whose order of arguments are consistent, and the file permissions are fine as is.

File: pkgs/development/python-modules/triton/default.nix (hunk 1 above)
Comment (pbsds) [reply pkgs/development/python-modules/triton/default.nix:181]: i based it on 

https://github.com/NixOS/nixpkgs/blob/58df299b0de456c0019c3fd465eca1556f2045fe/pkgs/development/python-modules/triton/default.nix#L156

File: pkgs/applications/video/kodi/addons/jellycon/default.nix (hunk 2 above)
Comment (pbsds) [reply pkgs/applications/video/kodi/addons/jellycon/default.nix:32]: do you consider this a blocker or are we good to merge?

Comment (Atemu) [reply pkgs/applications/video/kodi/addons/jellycon/default.nix:32]: Not at all.

**Pull request merged**